### PR TITLE
WIP! now it's possible to define property_path

### PR DESCRIPTION
**Pull request by oleg-andreyev**
Fixes https://github.com/dustin10/VichUploaderBundle/issues/1273

### CODE REVIEW

File: src/Form/Type/VichFileType.php [hunk 1]
```diff
@@ -214,4 +214,21 @@ protected function resolveDownloadLabel($downloadLabel, object $object, FormInte
 
         return ['download_label' => $downloadLabel];
     }
+
+    protected function getUploadableObject(FormInterface $form): ?object
+    {
+        if (!$parent = $form->getParent()) {
```
Comment (garak) [inline src/Form/Type/VichFileType.php:220]: Can we use `$parent !== $form->getParent()`?

File: src/Form/Type/VichFileType.php [hunk 2]
```diff
@@ -214,4 +214,21 @@ protected function resolveDownloadLabel($downloadLabel, object $object, FormInte
...
+            throw new \RuntimeException(__CLASS__ . ' cannot be used directly without Uploadable object');
+        }
+
+        $data = $parent->getData();
+        $propertyPath = $form->getConfig()->getOption('property_path');
+        if ($propertyPath) {
```
Comment (garak) [inline src/Form/Type/VichFileType.php:226]: is this a boolean? If not, please use the proper comparison

File: src/Form/Type/VichFileType.php [hunk 3]
```diff
@@ -214,4 +214,21 @@ protected function resolveDownloadLabel($downloadLabel, object $object, FormInte
...
+            // if property_path is defined, parent object should be Uploadable
+            $propertyPath = new PropertyPath($propertyPath);
+            return $this->propertyAccessor->getValue($data, $propertyPath->getParent());
```
Comment (garak) [inline src/Form/Type/VichFileType.php:229]: Missing blank line before `return`

**Review by garak — CHANGES_REQUESTED** [review]

**Comment by garak** [comment]
@oleg-andreyev did you have time to sort this out?

**Comment by oleg-andreyev** [comment]
> @oleg-andreyev did you have time to sort this out?

@garak sorry, no. But can try by end of this week

**Comment by VincentLanglet** [comment]
> > @oleg-andreyev did you have time to sort this out?
> 
> @garak sorry, no. But can try by end of this week

I can try to finish this pr if you don't have time ; tell me about it @oleg-andreyev

**Comment by oleg-andreyev** [comment]
Want to create functional test and will be done 👍

**Comment by VincentLanglet** [comment]
Any news ? Do you need help ?

**Comment by oleg-andreyev** [comment]
> Any news ? Do you need help ?

Yes, would be great. Sorry.

**Pull request closed without merging**